banks-server/src/banks_server.rs

```diff
@@ -331,6 +333,58 @@ impl Banks for BanksServer {
             .await
     }
 
+    async fn process_versioned_transaction_with_commitment_and_context(
+        self,
+        _: Context,
+        transaction: VersionedTransaction,
+        commitment: CommitmentLevel,
+    ) -> BanksTransactionResultWithMetadata {
+        let bank = self.bank(commitment);
+        let batch = match bank.prepare_entry_batch(vec![transaction]) {
+            Ok(batch) => batch,
+            Err(err) => {
+                return BanksTransactionResultWithMetadata {
+                    result: Err(err),
+                    metadata: None,
+                }
+            }
+        };
+
+        let (
+            TransactionResults {
+                mut execution_results,
+                ..
+            },
+            ..,
+        ) = bank.load_execute_and_commit_transactions(
+            &batch,
+            MAX_PROCESSING_AGE,
+            false, // collect balances
+            false, // enable cpi recording
+            true,  // enable log recording
+            false, // enable return data recording
```

Comment:
To go along with my earlier comment on `TransactionMetadata`, these will all need to be `true`